bazel/repository_locations.bzl

```diff
@@ -301,6 +301,33 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         release_date = "2020-08-25",
         cpe = "N/A",
     ),
+    skywalking_data_collect_protocol = dict(
+        project_name = "skywalking-data-collect-protocol",
+        project_desc = "Data Collect Protocols of Apache SkyWalking",
+        project_url = "https://github.com/apache/skywalking-data-collect-protocol",
+        name = "skywalking_data_collect_protocol",
+        sha256 = "fa9ac679624217f30b6e8d5c450365386c610e2d08188a20f0340c3b14401833",
+        urls = ["https://github.com/apache/skywalking-data-collect-protocol/archive/v8.3.0.zip"],
+        strip_prefix = "skywalking-data-collect-protocol-8.3.0",
+        version = "8.3.0",
+        use_category = ["observability_ext"],
+        extensions = ["envoy.tracers.skywalking"],
+        release_date = "2020-11-20",
+        cpe = "N/A",
+    ),
+    com_github_skyapm_cpp2sky = dict(
+        project_name = "cpp2sky",
+        project_desc = "C++ SDK for Apache SkyWalking",
```

Comment:
I had a look at the implementation of this library. It uses its own gRPC async client, built on Google gRPC SDK. I think there are two questions:
1. Does this async client play "nice" with the Envoy thread model? I.e. Do independent worker threads get their own completion queue threads, etc. We have spent some time investing in making this work well inside Envoy's Google gRPC async client.
2. This can only build when we don't have `--define google_grpc=disabled`.

IMHO ideally you'd be able to point your library at our async client and work even when there is no Google gRPC. This is not always possible in 3rd party libraries, but I'd encourage you to explore the practicality.

Reply:
I don't think this PR uses the Google gRPC SDK from cpp2sky but only the data model. The Google gRPC based one is more a generic feature for other C++ client using SkyWalking. We might need to be clear in the build target to make sure this doesn't pull google grpc as dependency.

Reply:
Yes. It uses only Skywalking's data definition. gRPC client abstraction has already accomplished by integration envoy internal grpc client. For more detail, https://github.com/envoyproxy/envoy/blob/a93219b052b6cd69f99dcb66f5c99c99b2cc072b/source/extensions/tracers/skywalking/trace_segment_reporter.cc

Reply:
I see, yep, makes sense to me then. I think as long as this is passing `compile_time_options` it probably works without gRPC. Adding some comments around this would be helpful.

Reply:
If you can add a comment here we can close this thread out. Other than this, I'm good to go based on your dependency analysis. @envoyproxy/dependency-shepherds any additional asks here?